src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -348,7 +348,7 @@ public HttpResponse doFinishLogin(final StaplerRequest request, final StaplerRes
         return HttpResponses.redirectTo(redirectUrl);
     }
 
-    private String getEffectiveLogoutUrl() {
+    private String getEffectiveLogoutUrl() throws NullPointerException {
```

Comment:
it is a runtime exception, the error is catastrophic, so it does not need to add the `throws`

Reply:
you should change Jenkins.getInstanceOrNull() to Jenkins.get(), I guess this method will not change anymore, there is a history about Jenkins.get(), Jenkins.getInstanceOrNull(), and Jenkins.getInstance()